agenthub/codeact_agent/prompt.py

```diff
@@ -0,0 +1,222 @@
+from opendevin.sandbox.plugins import SWEAgentCommandsRequirement
+
+_SWEAGENT_BASH_DOCS = '\n'.join(
+    filter(
+        lambda x: not x.startswith('submit'),
+        SWEAgentCommandsRequirement.documentation.split('\n')
+    )
+)
+# _SWEAGENT_BASH_DOCS content below:
+"""
+open <path> [<line_number>] - opens the file at the given path in the editor. If line_number is provided, the window will be move to include that line
+goto <line_number> - moves the window to show <line_number>
+scroll_down - moves the window down {WINDOW} lines
+scroll_up - moves the window down {WINDOW} lines
+create <filename> - creates and opens a new file with the given name
+search_dir <search_term> [<dir>] - searches for search_term in all files in dir. If dir is not provided, searches in the current directory
+search_file <search_term> [<file>] - searches for search_term in file. If file is not provided, searches in the current open file
+find_file <file_name> [<dir>] - finds all files with the given name in dir. If dir is not provided, searches in the current directory
+edit <start_line>:<end_line>
+<replacement_text>
+end_of_edit - replaces lines <start_line> through <end_line> (inclusive) with the given text in the open file. The replacement text is terminated by a line with only end_of_edit on it. All of the <replacement text> will be entered, so make sure your indentation is formatted properly. Python files will be checked for syntax errors after the edit. If the system detects a syntax error, the edit will not be executed. Simply try to edit the file again, but make sure to read the error message and modify the edit command you issue accordingly. Issuing the same command a second time will just lead to the same error message again.
+"""
+
+_COMMAND_DOCS = (
+    '\nApart from the standard bash commands, you can also use the following special commands in bash:\n'
+    f'{_SWEAGENT_BASH_DOCS}'
+    "Please note that THE EDIT COMMAND REQUIRES PROPER INDENTATION. If you'd like to add the line '        print(x)' you must fully write that out, with all those spaces before the code! Indentation is important and code that is not indented correctly will fail and require fixing before it can be run."
+)
+
+SYSTEM_MESSAGE = f"""A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions.
+The assistant can interact with an interactive Python (Jupyter Notebook) environment and receive the corresponding output when needed. The code should be enclosed using "<execute_ipython>" tag, for example: <execute_ipython> print("Hello World!") </execute_ipython>.
```

Comment:
This might not confuse LLM but it confused me 😛 . The example seems to imply it is fine to mess with indentation for `<execute_ipython>` tag. Note the space before and after the `print` statement in the old example.

```suggestion
The assistant can interact with an interactive Python (Jupyter Notebook) environment and receive the corresponding output when needed. The code should be enclosed using "<execute_ipython>" tag, for example: <execute_ipython>print("Hello World!")</execute_ipython>.
```

Reply:
Thanks for the suggestion!

I actually changed it to:
```
<execute_ipython>
print("Hello world!")
</execute_ipython>
```

Since, likely, putting `>p` together without space might potentially trigger some weird artifacts/issues in LLM sub-world tokenizers - just to be save :)

Reply:
yeah I think that's even better